public/app/features/dashboard/state/PanelQueryRunner.ts

```diff
@@ -295,3 +222,63 @@ async function getDataSource(
   }
   return await getDatasourceSrv().get(datasource as string, scopedVars);
 }
+
+function translateToLegacyData(data: DataQueryResponseData) {
+  return data.map((v: any) => {
+    if (isDataFrame(v)) {
+      return toLegacyResponseData(v);
+    }
+    return v;
+  });
+}
+
+/**
+ * All panels will be passed tables that have our best guess at colum type set
+ *
+ * This is also used by PanelChrome for snapshot support
+ */
+export function getProcessedDataFrames(results?: DataQueryResponseData[]): DataFrame[] {
+  if (!isArray(results)) {
+    return [];
+  }
+
+  const series: DataFrame[] = [];
+  for (const r of results) {
+    if (r) {
+      series.push(guessFieldTypes(toDataFrame(r)));
+    }
+  }
+
+  return series;
+}
+
+export function postProcessPanelData(format = PanelQueryRunnerFormat.frames, callback: (data: PanelData) => void) {
```

Comment:
Feels like transformations should belong here. 

`function postProcessPanelData(format, transformations) {`